src/docfx/lib/git/GitUtility.cs

```diff
@@ -213,6 +213,47 @@ public static void CheckMergeConflictMarker(string content, string file)
             }
         }
 
+        public static unsafe bool TryGetContentFromHistory(string repoPath, string filePath, string committish, out string content)
+        {
+            content = null;
+            if (git_repository_open(out var repo, repoPath) != 0)
+            {
+                return false;
+            }
+
+            if (git_revparse_single(out var commit, repo, committish) != 0)
```

Comment:
we only need to support commit hash for content retrieval, `revparse` might be slower/heavier then a simple lookup by git object id.

Reply:
i want to make it consistent with others and truly, we don't need do too much optimization for this scenario

---
In reply to: [236458528](https://github.com/dotnet/docfx/pull/3677#discussion_r236458528) [](ancestors = 236458528)